### DISCUSSION TITLE
Error: Cannot find module 'fs' after using it bundled with an npm package

### DISCUSSION THREAD

**Discussion by growupanand** [socketio/socket.io#5029] [Q&A]
I don't know what I'm missing or doing wrong, this is turborepo and the published package is internal package of workspace which is also using another unpublished package using socket.io.

I have published an npm package which was bundle with tsup. but when I am trying to install it another react app, I am getting below error. Issue of my project is here - https://github.com/growupanand/ConvoForm/issues/244.

Branch - https://github.com/growupanand/ConvoForm/tree/231-publish-react-hook-as-npm-package
NPM package link - https://www.npmjs.com/package/@convoform/react

```
Unhandled Runtime Error
Error: Cannot find module 'fs'

Call Stack
webpackEmptyContext
file:///home/growupanand/Development/repos/ConvoForm/apps/web/.next/static/chunks/app/(protectedPage)/forms/%5BformId%5D/page.js (18:10)
__require
../node_modules/.pnpm/engine.io-client@6.5.3/node_modules/engine.io-client/build/cjs/contrib/yeast.js (46:9)
__require2
(app-pages-browser)/../../packages/react/dist/index.js
require_XMLHttpRequest
../node_modules/.pnpm/engine.io-client@6.5.3/node_modules/engine.io-client/build/cjs/transports/xmlhttprequest.js (15:21)
__require2
(app-pages-browser)/../../packages/react/dist/index.js
require_xmlhttprequest
../node_modules/.pnpm/engine.io-client@6.5.3/node_modules/engine.io-client/build/cjs/transports/xmlhttprequest.js (109:5)
__require2
(app-pages-browser)/../../packages/react/dist/index.js
require_polling
../node_modules/.pnpm/engine.io-client@6.5.3/node_modules/engine.io-client/build/cjs/transports/webtransport.js (58:26)
__require2
(app-pages-browser)/../../packages/react/dist/index.js
require_transports
../node_modules/.pnpm/engine.io-client@6.5.3/node_modules/engine.io-client/build/cjs/contrib/parseuri.js (42:13)
```

**Screenshots**
![image](https://github.com/growupanand/ConvoForm/assets/29487686/cc8e3434-76e8-4eff-b87b-25e98d1a9301)

**Comment by darrachequesne** [comment]
Hi!

That's weird, it seems the bundle uses the Node.js implementation of the HTTP long-polling transport (based on the `xmlhttprequest-ssl` package) and not the browser one (which uses the native `XMLHttpRequest` object).

The selection should be made with the `browser` field there: https://github.com/socketio/engine.io-client/blob/62cbed729de418c85d89283e2f196df26d47aaff/package.json#L107

**Reply by growupanand** [reply]
Hi,
Thanks for reply, do you think any solution for this, because I am stuck at this from last week. Let me know If you need my help, I am available for call also to show you this on my machine.

**Comment by growupanand** [comment]
It is fixed, I have a turborepo, 

there is internal package `@convoform/websocket-client` which is using `socket.io-client`

there is internal package `@convoform/react` which is using `@convoform/websocket-client` and I want to publish this package only.

**Problem**
I was trying to first build `@convoform/websocket-client` with `socket.io-client`
then build `@convoform/react` with `@convoform/websocket-client`

**Solution**
Do not need to build `@convoform/websocket-client`, just use it as dev dependency `"@convoform/websocket-client": "../websocket-client"`
